src/tls/cockpit-certificate-ensure.c

```diff
@@ -0,0 +1,64 @@
+#include <assert.h>
+#include <err.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include <common/cockpitwebcertificate.h>
+
+#include "certificate.h"
+#include "utils.h"
+
+// Renew certificates with less than 30 days validity
+#define EXPIRY_THRESHOLD (30 * 24 * 60 * 60)
+
+static bool
+check_expiry (const char *filename)
+{
+  Certificate *certificate = certificate_load (filename);
```

Comment:
Note to self: `certificate_load()` aborts if file can't be loaded. Thus no error handling necessary here.